great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -1,17 +1,67 @@
-from typing import Literal, Union
+from __future__ import annotations
 
-from pydantic import AnyUrl
+from typing import TYPE_CHECKING, Literal, Optional, Union
 
+import pydantic
+from pydantic import AnyUrl, errors
+
+from great_expectations.compatibility.snowflake import URL
+from great_expectations.compatibility.sqlalchemy import sqlalchemy as sa
 from great_expectations.core._docs_decorators import public_api
 from great_expectations.datasource.fluent.config_str import ConfigStr
-from great_expectations.datasource.fluent.sql_datasource import SQLDatasource
+from great_expectations.datasource.fluent.sql_datasource import (
+    SQLDatasource,
+    SQLDatasourceError,
+)
+
+if TYPE_CHECKING:
+    from pydantic.networks import Parts
+
+    from great_expectations.compatibility import sqlalchemy
+
+
+class _UrlPasswordError(pydantic.UrlError):
```

Comment:
I have not seen this convention of `_ClassName` before, what does the prefix underscore mean or indicate?

Reply:
just that it is internal - we don't have access modifiers in Python so we've used this in places to denote that it shouldn't be used by other modules

Reply:
updates to docs will come in a later effort as part of this epic!